### PR TITLE
[FIX] mass_mailing: handle UnidentifiedImageError while convert image to url

### PR DESCRIPTION
**Pull request by maan-odoo**
Currently, a logger exception is generated when the user tries to upload any document in the mass mail.

Stack trace on sentry:
```
UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7f1ad928db70>
  File "addons/mass_mailing/models/mailing.py", line 1437, in _get_image_by_url
    image = Image.open(io.BytesIO(content))
  File "PIL/Image.py", line 3008, in open
    raise UnidentifiedImageError(
```

This is because an UnidentifiedImageError occurs when the user uploads
an image file as a document and code [1] tries to open it with Image.

This commit adds code that handles an UnidentifiedImageError, and it adds
the message in the log for an invalid image file.

[1]-https://github.com/odoo/odoo/blob/029b84f3c061f819bacb9a4818504cced4adeb1c/addons/mass_mailing/models/mailing.py#L1405

sentry-4311184876

### CODE REVIEW

**Comment by maan-odoo** [comment]
Hello @odoo/sentry_reviewers 
Please review this PR.

Thanks!

**Comment by maan-odoo** [comment]
Hello @tde-banana-odoo 
Please look at this small PR!
Thanks!

**Comment by tde-banana-odoo** [comment]
But then message "Could not retrieve URL" is not correct. Can't you catch this specific exception and write a nicer error message in that case ?

**Comment by maan-odoo** [comment]
> But then message "Could not retrieve URL" is not correct. Can't you catch this specific exception and write a nicer error message in that case ?

Hello @tde-banana-odoo 
I have updated the code now that it handles UnidentifiedImageError and writes a log warning message. 

Thanks! :slightly_smiling_face:

File: addons/mass_mailing/models/mailing.py [hunk 1]
```diff
@@ -1410,6 +1410,9 @@ def _get_image_by_url(self, url, session):
                 )
 
             return content
+        except UnidentifiedImageError as e:
+            _logger.warning('This file could not be decoded as an image file.', exc_info=True)
+            raise e
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/models/mailing.py:1415]:
```suggestion
            raise
```

Comment (maan-odoo) [reply addons/mass_mailing/models/mailing.py]: Done

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**